modules/cert-manager-acme-dns01-explicit-gcp.adoc

```diff
@@ -0,0 +1,150 @@
+// Module included in the following assemblies:
+//
+// * security/cert_manager_operator/cert-manager-operator-issuer-acme.adoc
+
+:_content-type: PROCEDURE
+[id="cert-manager-acme-dns01-explicit-gcp_{context}"]
+= Configuring an ACME issuer to solve DNS-01 challenges by using explicit credentials for GCP Cloud DNS
+
+You can use the {cert-manager-operator} to set up an ACME issuer to solve DNS-01 challenges by using explicit credentials on GCP. This procedure uses _Let's Encrypt_ as the ACME CA server and shows how to solve DNS-01 challenges with Google CloudDNS.
+
+.Prerequisites
+
+* You have access to a role that has permissions to create issuers, certificates, and secrets in the desired namespace.
+* You have access to the {product-title} cluster as a user with the `cluster-admin` role.
+* You have set up an IAM role for Google CloudDNS. For more information, see link:https://cert-manager.io/docs/configuration/acme/dns01/google/[Google CloudDNS] in the upstream cert-manager documentation.
++
+[NOTE]
+====
+You can use Google CloudDNS with explicit credentials in an {product-title} cluster that is not running on GCP.
+====
+
+.Procedure
+
+. Optional: Override the nameserver settings for the DNS-01 self check:
++
+This step is required only when the target public-hosted zone overlaps with the cluster's default private-hosted zone.
+
+.. Edit the `CertManager` resource by running the following command:
++
+[source,terminal]
+----
+$ oc edit certmanager cluster
+----
+
+.. Add a `spec.controllerConfig` section with the following override arguments:
++
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1alpha1
+kind: CertManager
+metadata:
+  name: cluster
+  ...
+spec:
+  ...
+  controllerConfig:                                <1>
+    overrideArgs:
+      - '--dns01-recursive-nameservers-only'       <2>
+----
+<1> Add the `spec.controllerConfig` section.
+<2> Specify to only use recursive nameservers instead of checking the authoritative nameservers associated with that domain.
+
```

Comment:
This is incorrect. There are 3 occurrences where it only documents --dns01-recursive-nameservers-only without --dns01-recursive-nameservers=1.1.1.1:53. All occurrences should document both.

Reply:
But this is not a suggested documenting way. Reading https://59702--docspreview.netlify.app/openshift-enterprise/latest/security/cert_manager_operator/cert-manager-operator-issuer-acme.html#cert-manager-acme-dns01-explicit-azure_cert-manager-operator-issuer-acme , it shows 5 occurrences of `Optional: Override the nameserver ...`, it is redundant, duplicate and easy to be inconsistent as pointed out.
Suggest please not duplicate and just put this in one occurrence somewhere for example at the beginning of all dns-01 sections.

Reply:
@xingxingxia So considering this structure, right now we cannot avoid those redundant instructions. Because for eg, if a customer jumps on any particular section/module he will follow the sequential steps without referring to any other section. So we must provide step-by-step instructions. To avoid duplicate steps I can think of restructuring but that might take some time. We can get this structure merged and then I can work on restructuring OR if we can wait then I can start working on restructuring immediately.

Reply:
I'd suggest that for the time being we keep the current structure, cause though it contains some redundant information it is informative for users with completeness following particular sub sections to achieve what they're looking for through the steps in the docs exclusively, as long as we have correctness of the functionality. IMO, we probably could focus on getting this out and later we can revisit to remove some of the redundancies through plausible restructuring.

Reply:
OK, it is fine, but please ensure `All occurrences should document both`. Now there is still one occurrence under `ambient credentials on GCP` section which still only documents --dns01-recursive-nameservers-only without --dns01-recursive-nameservers=1.1.1.1:53 .

Reply:
Copying and pasting would work, but one nagging here, not sure why some some occurrences of same snippets are:
```
      - '--dns01-recursive-nameservers-only'       
      - '--dns01-recursive-nameservers=1.1.1.1:53'
```
While other occurrences are inconsistent order:
```
      - '--dns01-recursive-nameservers=1.1.1.1:53' 
      - '--dns01-recursive-nameservers-only'  
```